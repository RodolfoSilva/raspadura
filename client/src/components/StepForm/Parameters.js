import React from 'react';
import { Field } from 'redux-form';

export default class Parameters extends React.Component {

  render() {
    const { fields, meta: { error } } = this.props;
    return (
      <div>
        <button type="button" onClick={() => fields.push({})}>Adicionar parametro</button>
        <ul>
          {fields.map((property, index) => (
            <li key={index}>
              <button type="button" onClick={() => fields.remove(index)}> X </button>
              <Field
                name={`${property}.name`}
                type="text"
                component="input"
                label={`Hobby #${index + 1}`}
              />
              <Field
                name={`${property}.value`}
                type="text"
                component="input"
              />
            </li>
          ))}
          {error && <li className="error">{error}</li>}
        </ul>
      </div>
    );
  }
}