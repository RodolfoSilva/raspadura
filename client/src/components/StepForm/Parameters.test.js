import React from 'react';
import { shallow } from 'enzyme';
import Parameters from './Parameters';

describe('Parameters', () => {
  it('render empty', () => {
    const wrapper = shallow(<Parameters fields={[]} meta={{ error: null }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render with values', () => {
    const fields = [
      { name: 'code', value: '1324564' }
    ];
    const wrapper = shallow(<Parameters fields={fields} meta={{ error: null }} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('add a new field', () => {
    const fields = [
      { name: 'code', value: '1324564' }
    ];
    const wrapper = shallow(<Parameters fields={fields} meta={{ error: null }} />);

    wrapper.find('button').first().simulate('click');
    expect(fields).toHaveLength(2);

    wrapper.instance().forceUpdate();
    expect(wrapper.update()).toMatchSnapshot();
  });
});