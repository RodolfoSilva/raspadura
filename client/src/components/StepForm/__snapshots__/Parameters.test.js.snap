// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parameters add a new field 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Adicionar parametro
  </button>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
         X 
      </button>
      <Field
        component="input"
        label="Hobby #1"
        name="[object Object].name"
        type="text"
      />
      <Field
        component="input"
        name="[object Object].value"
        type="text"
      />
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
         X 
      </button>
      <Field
        component="input"
        label="Hobby #2"
        name="[object Object].name"
        type="text"
      />
      <Field
        component="input"
        name="[object Object].value"
        type="text"
      />
    </li>
  </ul>
</div>
`;

exports[`Parameters render empty 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Adicionar parametro
  </button>
  <ul />
</div>
`;

exports[`Parameters render with values 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Adicionar parametro
  </button>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
         X 
      </button>
      <Field
        component="input"
        label="Hobby #1"
        name="[object Object].name"
        type="text"
      />
      <Field
        component="input"
        name="[object Object].value"
        type="text"
      />
    </li>
  </ul>
</div>
`;
